superset/charts/data/api.py

```diff
@@ -299,6 +299,9 @@ def data_from_cache(self, cache_key: str) -> Response:
         """
         try:
             cached_data = self._load_query_context_form_from_cache(cache_key)
+            
+            # Set form_data in Flask Global as it is used as a fallback for async queries with jinja context
+            setattr(g, "form_data", cached_data)
```

Comment:
you appear to be missing an import
```python
from flask import g
```

Reply:
thanks! missed that, was running a different version locally.